src/modules/auth/auth.module.ts

```diff
@@ -46,12 +49,28 @@ import { STATUS_LIST_MODULE_PATH } from '../status-list/status-list.const';
   exports: [JwtAuthGuard, JwtStrategy, GqlAuthGuard],
 })
 export class AuthModule implements NestModule {
-  constructor(private readonly tokenService: TokenService) {}
+  private allowedOrigins: string[];
+
+  constructor(
+    private readonly tokenService: TokenService,
+    private readonly configService: ConfigService
+  ) {
+    this.allowedOrigins = JSON.parse(this.configService.get('ALLOWED_ORIGINS'));
+    for (const origin of this.allowedOrigins) {
+      if (!isURL(origin)) {
+        throw new Error(`Origin ${origin} is not a valid URL`);
+      }
+    }
+  }
 
   configure(consumer: MiddlewareConsumer) {
     consumer
-      .apply((req, res, next) =>
-        this.tokenService.handleOriginCheck(req, res, next)
+      .apply((req: Request, res: Response, next: NextFunction) =>
+        this.checkAllowedOrigin(req, next)
+      )
+      .forRoutes({ path: '/*', method: RequestMethod.ALL })
```

Comment:
I agree. Though `app.enableCors` is more consistent with how we adding other middleware, then `NestFactory.create`. We are calling methods on `app`

Reply:
Ah okay, I didn't realize that `app.enableCors` and the `cors` object in `NestFactory.create` are the same thing but that makes sense. You're right, it's better to modify the line that already has `app.enableCors` rather than switch the pattern we're using https://github.com/energywebfoundation/ssi-hub/blob/605820d672392b0ee72c107c2d150aef39d9b4c1/src/main.ts#L73